sklearn/feature_selection/_rfe.py

```diff
@@ -289,6 +290,15 @@ def _fit(self, X, y, step_score=None, **fit_params):
         else:  # float
             n_features_to_select = int(n_features * self.n_features_to_select)
 
+        if n_features_to_select > n_features:
+            warnings.warn(
+                (
+                    "features_to_select shouldn't be greater than the number of"
+                    " available features"
```

Comment:
We need to mention the actual behavior, what about something similar to

`n_features_to_select must not exceed available features. Falling back to select all n_features.`